cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -123,4 +124,24 @@ int64 truncate_int64_int32(int64 in, int32 out_scale) {
       gandiva::BasicDecimalScalar128(decimal_with_outscale, 38, out_scale), &overflow);
 }
 
+FORCE_INLINE
+float64 random_internal() {
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_real_distribution<> dis(0.0, 1.0);
+  return dis(gen);
+}
+
+FORCE_INLINE
+float64 random_int32_internal(int32 seed, bool seed_validity) {
+  if (!seed_validity) {
+    seed = 0;
+  }
+  int64 seed64 = static_cast<int64>(seed);
```

Comment:
could you add a comment describing what this seed manipulation accomplishes?

Reply:
I wanted to scramble the seed(x) in such a way that the result is a positive number and is not the same for x or -x. I looked up how java.util.Random scrambles the seed; lifted the constants from there